subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -473,6 +1060,7 @@ static int fp_storage_ak_reset(void)
 {
 	int err;
 	bool was_enabled = is_enabled;
+	const struct fp_storage_ak_bond_bt_request_cb *registered_bt_request_cb = bt_request_cb;
```

Comment:
I think we shouldn't reset the callback during factory reset to preserve the behaviour from other Fast Pair modules. We can consider them as a special RAM object that is not affected by factory reset. BTW, if you get an error in the `fp_bond_settings_delete`, you will lose the callback context.

Reply:
We reinstate it after ram_clear. Here we save it for later to call fp_storage_ak_bond_bt_request_cb_register

Reply:
I should call `fp_storage_ak_bond_bt_request_cb_register(registered_bt_request_cb);` after `fp_storage_ak_ram_clear` regardless of the `was_enabled` variable. Then it will be fine.

Reply:
discussed offline

```
if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
	registered_bt_request_cb = bt_request_cb;
}

fp_storage_ak_ram_clear();

if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
	bt_request_cb = registered_bt_request_cb;
}
```